src/cursor/cur_table.c

```diff
@@ -623,6 +623,29 @@ err:	CURSOR_UPDATE_API_END(session, ret);
 }
 
 /*
+ * __curtable_reserve --
+ *	WT_CURSOR->reserve method for the table cursor type.
+ */
+static int
+__curtable_reserve(WT_CURSOR *cursor)
+{
+	WT_CURSOR_TABLE *ctable;
+	WT_DECL_RET;
+	WT_SESSION_IMPL *session;
+
+	ctable = (WT_CURSOR_TABLE *)cursor;
+	JOINABLE_CURSOR_UPDATE_API_CALL(cursor, session, update, NULL);
+	WT_ERR(__curtable_open_indices(ctable));
+
+	/* Reserve in the primary, ignore the indices. */
+	APPLY_CG(ctable, reserve);
```

Comment:
In tables, `primary` means the first column group, i.e., `ctable->cg_cursors[0]`.

We could consider doing what the comment says, and only reserving in one column group.  That will only make a difference in complex tables, and the downside is if we one day get smarter about updates to complex tables (only applying the update to column groups that have changed), then primary-only reserve would not be sufficient to guarantee subsequent updates to the record will succeed.

TL;DR this looks right to me.